server/src/test/java/org/elasticsearch/action/ingest/SimulatePipelineRequestParsingTests.java

```diff
@@ -211,6 +211,38 @@ public void testParseWithProvidedPipeline() throws Exception {
         assertThat(actualRequest.pipeline().getProcessors().size(), equalTo(numProcessors));
     }
 
+    public void testParseWithNestedProcessors() throws Exception {
+        Map<String, Object> requestContent = new HashMap<>();
+        Map<String, Object> pipelineConfig = new HashMap<>();
+        List<Map<String, Object>> processors = new ArrayList<>();
+        Map<String, Object> processorConfig = new HashMap<>();
```

Comment:
I'd declare the `processorConfig` right before you use it, slightly clearer that way.